testing/deployer/topology/topology.go

```diff
@@ -422,9 +476,18 @@ const (
 	NodeKindDataplane NodeKind = "dataplane"
 )
 
+type NodeVersion string
+
+const (
+	NodeVersionUnknown NodeVersion = ""
+	NodeVersionV1      NodeVersion = "v1"
+	NodeVersionV2      NodeVersion = "v2"
+)
+
 // TODO: rename pod
 type Node struct {
 	Kind      NodeKind
+	Version   NodeVersion
```

Comment:
Is it possible to combine this with the `Kind` field? Are all combinations of different Kind and Version valid?

Reply:
They are unrelated concepts, as ideally v1/v2 feature parity will mean that all of the node kinds that work in v1 will be representable in v2 eventually.

But for the 1.17 target only non-peered/non-federated/gateway-less/dataplane-only clusters can be defined in v2.

Things like mesh gateways are not representable in v2 yet, but they should be in 1.18/1.19.

Client agents may need to ultimately be "hollowed out" at some point so that they can continue to exist, but use v2 endpoints.

As it's a different axis of definition it seemed more appropriate to make it a separate attribute of the node definition than trying to conflate the concepts into kind.